Blog - basic and more...

alexvoss opened the pull request:
Produced a basic example of the use of the blog plugin. Added use of authors file because I imagine a lot of people will want this. 

Will produce some more complex examples and as part of this will probably revisit this one, so not asking for a review yet. Feel free to comment, though.

Also, I created a little `meta.yml` file. Not sure if you had already thought about creating the index pages. Hope this is useful, if not LMK.

alexvoss committed:
an initial idea about how we might manage a list of examples

Thought adding metadata to each example would make this easier
to maintain than if we had to separately add things to a list.
Happy to have a go at writing something to compile the list of
examples. Now that I write this, I seem to remember there is a
plugin - mkdocs-gen-files?

alexvoss committed:
fixed stupid cut-n-paste error, using # for CSS comments...

alexvoss committed:
first stab at a basic blog example

alexvoss committed:
more ideas about what the metadata could look like

alexvoss committed:
added missing requirements.txt

alexvoss committed:
added an example for adding RSS to a blog

alexvoss commented:
Ok, the name of the branch is out of date - I added the Blog+RSS example just now. Requesting a review while I have my afternoon nap ;o)

alexvoss commented:
Before I forget, the one other use case I had on my list was a stand-along blog (as described in your blog post announcing the plugin, right?) - any other ones?

squidfunk commented on examples/blog-basic/docs/blog/posts/2023-10-12-welcome.md:
```diff
@@ -0,0 +1,32 @@
+  - hello
+---
+
+# Welcome to Material for MkDocs
```
For the sake of consistency and to better show time-based ordering I think it might be a better idea to just call this "My second blog post" 😉

alexvoss replied:
called it "a second post" because the first, more poetic one was written by me!

squidfunk commented on examples/blog-basic/meta.yml:
```diff
@@ -0,0 +1,16 @@
+meta:
+  tags:
+    - public        # works without Insiders
+      #- insiders      # requires Insiders edition
```
Why the commented out tags? Tags were not used in any blog post. I understand that this is to configure the RSS plugin, but especially "insiders" + "requires the Insiders edition" might be confusing to readers.

alexvoss replied:
Ah, sorry, caused a misunderstanding here. The meta.yml contains information I thought would be useful to build index pages for the examples. LMK if you no like and I remove.

squidfunk replied:
We should only include in the examples what is absolutely necessary and examples should be self-contained. In order to build the examples repository itself, we alter the configurations before building them using the [`projects_config_transform`](https://squidfunk.github.io/mkdocs-material/plugins/projects/#config.projects_config_transform) setting. We should add all stuff that only we need to build the examples and to generate indexes here:

https://github.com/mkdocs-material/examples/blob/cc63d197da71a7986776b5d2771d077b282adb40/examples/__init__.py#L31-L56

squidfunk replied:
Currently, the `transform` function does the following:

- Set the repo URL to the folder of the example
- Set the edit URI accordingly
- Set the copyright form the top-level `mkdocs.yml`
- Enable some features from the top-level `mkdocs.yml`
- Add icon configuration from the top-level `mkdocs.yml`

alexvoss replied:
ok - but where should the metadata go that describes the examples? We could put the existing `meta.yml` files together and store the data under `examples/examples/meta.yml` - or anything else you suggest but I would like to start doing it, otherwise it will be a lot of reverse-engineering later on.

squidfunk replied:
Okay, now I understand the use case, pardon my stupidity. I think we should add them to the root of each example next to`mkdocs.yml`, and ensure that they are not bundled with the downloadable example zips. We could then consume them in our transform function and/or a hook. Maybe we can start by calling it `.example.yml`, then it's clear that it's not MkDocs-related.

squidfunk commented on examples/blog-basic/mkdocs.yml:
```diff
@@ -0,0 +1,42 @@
+  name: material
+
+# Plugins 
+# Note that you need to add the search plugin manually
```
I'd rather not mention this, as we will have to mention it in each and every `mkdocs.yml` file then. Just add the search and let the user find out what happens when they remove it.

squidfunk commented on examples/blog-basic/mkdocs.yml:
```diff
@@ -0,0 +1,42 @@
+plugins:
+  - search
+  - blog:
+      pagination: true
```
Both settings can be removed, as we only have 2 blog posts (with a default of 10 per page) and pagination is enabled by default. I'd say we keep it as minimal as possible.

squidfunk commented on examples/blog-rss/mkdocs.yml:
```diff
@@ -0,0 +1,56 @@
+
+# Plugins
+plugins:
+  # Note that you need to add the search plugin manually
```
All other comments also apply to the `blog-rss` example.

squidfunk commented on examples/page-status/mkdocs.yml:
```diff
@@ -1,15 +1,46 @@
+# Copyright (c) 2016-2023 the authors:
```
Please adhere to the format 😉

```
# Copyright (c) 2016-2023 Martin Donath <martin.donath@squidfunk.com>
                          Alex Voss <alex@corealisation.com>
```

squidfunk commented:
Oh, and I'd drop the date from the filenames or users might think that it is mandatory.

alexvoss committed:
incorporated PR feedback

- renamed and tidied up example metadata
- changed formatting of copyright
- renamed blog post files
- retitled 2nd blog post
- removed unnecessary stuff from mkdocs.yml
- for both examples

alexvoss commented:
Just realised the RSS is causing a warning because it is not in the nav - will try and fix. 

Edit: right, I am having no joy with the `not_in_nav` functionality. Am wondering if deleting the link from README.md would be the way to go but how do I point people at the RSS feed? I know that you can just point an RSS feed reader to the site but IMHO it would be good if there was some UI element for the feed to advertise it.

alexvoss commented:
Also, let me know if you want me to un-draft this once the above question is resolved so you can merge. I still need to do the blog-only site but need to do a couple of other things now.

squidfunk commented:
Already undrafted it, as I thought you were ready. Once I approved it, you can also merge it (I think you should be able to, being a collaborator). If you feel that you made changes that need re-review, just re-request it ☺️ I'm going to groom the examples from time to time, but for the time being you can go ahead and merge it if approved.

squidfunk committed:
Merge branch 'master' into blog-basic

alexvoss committed:
removed link to feed from README...

and added a note about how to use it with an RSS
reader. I would still love to have a UI element for
accessing the feed

alexvoss committed:
reformatted copyright notice
